neonvm-kernel/README.md

```diff
@@ -15,29 +15,33 @@ in use, although this can be overridden on an individual VM basis using the
 
 ## Upgrading the kernel
 
+NB: upgrading kernel config should be done for both supported architectures.
 Assuming a plain upgrade (i.e. no additional features to enable), upgrading the kernel can be done
 with the following sequence of actions:
 
 1. On the host, run:
    ```sh
+   export OLD_VERSION=<replace with version>
+   export NEW_VERSION=<replace with version>
+   export KERNEL_ARCH=amd64 # replace with the wanted arch, x86_64 or arm64
    cd neonvm-kernel # this directory
-   docker build --build-arg KERNEL_VERSION=$NEW_VERSION --platform linux/x86_64 --target build-deps -t kernel-build-deps -f Dockerfile.kernel-builder .
+   docker build --build-arg KERNEL_ARCH=$ARCH --build-arg NEW_VERSION=$NEW_VERSION --build-arg OLD_VERSION=$OLD_VERSION --platform linux/x86_64 --target build-deps -t kernel-build-deps -f Dockerfile.kernel-builder .
    docker run --rm -v $PWD:/host --name kernel-build -it kernel-build-deps bash
    ```
 2. Then, inside the container, run:
    ```sh
    cd linux-$NEW_VERSION
-   cp /host/linux-config-6.1.92 .config # Copy current config in
-   make menuconfig
+   cp /host/linux-config-$ARCH-$OLD_VERSION .config # Copy current config in
+   make ARCH=$KERNEL_ARCH menuconfig
```

Comment:
If these environment variables are defined on the host, are they actually available inside the container?

Reply:
Oh, I passed them as a build-arg 🤦 Moved to the docker run -e